src/google/protobuf/compiler/objectivec/objectivec_helpers.cc

```diff
@@ -868,7 +870,12 @@ string BuildCommentsString(const SourceLocation& location,
     // Decouple / from * to not have inline comments inside comments.
     line = StringReplace(line, "/*", "/\\*", true);
     line = StringReplace(line, "*/", "*\\/", true);
-    final_comments += prefix + line + suffix;
+    line = prefix + line;
+    StripWhitespace(&line);
+    // If not a one line, need to add the first space before *, as
+    // StripWhitespace would have removed it.
+    line = (is_one_line ? "" : " ") + line;
+    final_comments += line + suffix;
```

Comment:
What is we drop the \n from both suffixes, then I think you can just do:

line = prefix + line + suffix
StripWhiteSpace(&line)
final_comments += line + "\n"

And there's isn't the special knowledge about oneline needing the space back.